backend/src/core/services/stats.service.ts

```diff
@@ -154,6 +154,73 @@ const getCurrentStats = async (
     halted: job.campaign.halted,
     status_updated_at: job.updatedAt, // Timestamp when job was logged
   }
+
+  // If job is not in send queue, return archived stats
+  if (job.status !== JobStatus.Ready) {
+    return stats
+  }
+
+  const campaign = await Campaign.findByPk(campaignId)
+  if (!campaign || !campaign.credName) {
+    throw new Error('Unable to find campaign in campaigns table.')
+  }
+
+  // check if there exists a job in progress using this credential
+  const runningJob = await JobQueue.findOne({
+    where: { workerId: { [Op.ne]: null } },
+    include: [
+      {
+        model: Campaign,
+        where: { credName: campaign.credName },
+        include: [Statistic],
+      },
+    ],
+  })
+
+  // get all jobs in send queue using this credential
+  const sharedCredentialJobs = await JobQueue.findAll({
+    where: { status: JobStatus.Ready },
+    include: [
+      {
+        model: Campaign,
+        where: { credName: campaign.credName },
+        include: [Statistic],
+      },
+    ],
+  })
+
+  // get the job's posiiton in the queue
+  const position = await sharedCredentialJobs.findIndex(
+    (job) => job.campaignId === campaignId
+  )
+
+  // if there are no existing jobs running and this job is the first in the send queue
+  // no need to send queue status
+  if (!runningJob && position === 0) {
+    return stats
+  }
+
+  // get all jobs in queue before this job
+  const jobsInQueue = sharedCredentialJobs.slice(0, position)
+  if (runningJob) jobsInQueue.push(runningJob)
+
+  // use number of unsent messages from statistics and job send_rate
+  // to compute approx time needed to complete jobs
+  let totalTime = 0
+  jobsInQueue.map((job) => {
+    if (job.campaign.statistic?.unsent && job.sendRate) {
+      const timeCurrJob = job.campaign.statistic.unsent / job.sendRate
+      totalTime += timeCurrJob
+    }
+  })
+
+  return {
+    ...stats,
+    queueStatus: {
+      position,
```

Comment:
Position can be found based on the updated_at of the existing jobs which use the same credential. But actually, knowing the position isn't important right? The user just needs to know the wait time.

Maybe something like: 
Get the updated_at and cred_name for this campaign
```sql
SELECT j.updated_at, c.cred_name FROM job_queue  j , campaigns c WHERE 
	j.campaign_id = :campaign_id 
	AND c.id = j.campaign_id
```

Calculate the wait time.
```sql
SELECT SUM(s.unsent / j.send_rate) AS wait_time FROM job_queue j, campaigns c, statistics s WHERE 
c.cred_name = :cred_name
AND s.campaign_id = c.id
AND j.campaign_id = c.id 
AND j.status = 'READY' 
AND j.updated_at < :updated_at;
```